### DISCUSSION TITLE
Wiesław Šoltés SvgToXaml: Method not found: 'Void SkiaSharp.SKPath.Transform(SkiaSharp.SKMatrix)'

### DISCUSSION THREAD

**Discussion by Yveske1966** [AvaloniaUI/Avalonia#15065] [Q&A] [answered]
I've got the project running once before, but after upgrading it stopped working.
The project compiles and runs up to the point it needs to evaluate an SVG.

    public SkiaSharp.SKPicture? FromSvg(string svg, DrawAttributes ignoreAttributes)
    {
        Reset();
        var svgDocument = SvgExtensions.FromSvg(svg);
        if (svgDocument is { })
        {
            Model = SvgExtensions.ToModel(svgDocument, s_assetLoader, out var drawable, out _, ignoreAttributes);
            Drawable = drawable;
            Picture = s_model.ToSKPicture(Model);   <<<< Here !!!
            return Picture;
        }
        return null;
    }

I filed an issue with Wiesław Šoltés on March 7th (https://github.com/wieslawsoltes/SvgToXaml/issues/16), but got no response.
As of today I'm using:
Using Visual Studio 17.10.0 Preview 2.0 (latest)
Updated project to target Net8.0 and Windows 10.0.22621.0 (latest)
Updated Avalonia Nuget packages to 11.1.0-beta1 (latest)
Updated Avalonia.Controls.Skia package to 11.1.0-beta1 (latest)
Updated Svg.Skia package to 2.0.0-beta1 (latest)

Am I missing something (a package, a usage, ...) ?

**Comment by thevortexcloud** [comment, marked as answer]
Method not found is typically a dependency issue. .NET does a "shoot first ask questions later" approach to determining if a method actually exists. As such it will attempt to load a library it thinks it was built against and then try to invoke code from it, even if the code it's looking for no longer exists due to the DLL being replaced with another version. My guess is you need to downgrade/upgrade a package. Working out which one can be tricky. Your best bet is to work out what has a dependency on Skia and what version of Skia they require. It's possible some of the packages are using different versions which is causing a subtle conflict.

**Reply by Yveske1966** [reply]
Thanks ! Will have my work cut out for me then...
Will try to find the solution and come back later to mark this as an answer.

**Reply by Yveske1966** [reply]
Ok, downloading again and not updating makes it work again, but the issue should be addressed in the future.

**Comment by maxkatz6** [comment]
`Method not found: 'Void SkiaSharp.SKPath.Transform(SkiaSharp.SKMatrix)'` looks like there is a mixed SkiaSharp version. I.e. SkiaSharp 3.0 used together with 2.88.

**Reply by maxkatz6** [reply]
11.1.0-beta1 should work with both versions, but I would recommend to stick to 2.88 for now.
